### PR TITLE
Added Abstraction to twitter API to allow for a pluggable backend. cu…

### PR DESCRIPTION
**Pull request by mklaehn**
…rrently only twitter backend. moved to abstraction in all projects previously using twitter4j library. fixes JUGBodensee/TweetwallFX#15

### CODE REVIEW

File: 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java [hunk 1]
```diff
@@ -149,15 +132,9 @@ public void start() {
...
-            query.setCount(100);
-            twitter = new TwitterFactory(conf).getInstance();
-            QueryResult result = twitter.search(query);
-
-            buildTagCloud(result.getTweets());
+            buildTagCloud(tweeter.search(new TweetQuery().query(searchText).count(100)));
             createWordle();
-
-        } catch (TwitterException ex) {
+        } catch (TweetException ex) {
```
Comment (svenreimers) [inline 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java]: Enhance exception handling by providing a better error facade in new tweet api

File: build.gradle [hunk 2]
```diff
@@ -38,19 +38,41 @@ subprojects {
...
+    }
+}
+
+project(":TweetWallFX-Tweet-API") {
+    dependencies {
+        compile group: 'org.netbeans.api', name: 'org-openide-util-lookup', version: 'RELEASE80'
+    }
+}
+
+project(":TweetWallFX-Tweet-SPI-Twitter") {
```
Comment (svenreimers) [inline build.gradle]: Is spi-twitter the correct naming? Probably impl-twitter4j would be more exact. SPI should be reserved for an spi not an implementation.

File: 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java [hunk 3]
```diff
@@ -177,25 +154,23 @@ public void stop() {
...
-        private TwitterStream stream;
-        private final Configuration conf;
-        private final BlockingQueue<TweetInfo> tweets;
+        private TweetStream stream;
+        private final Tweeter tweeter;
+        private final BlockingQueue<Tweet> tweets;
 
-        public TweetsCreationTask(Configuration conf, String searchText, BlockingQueue<TweetInfo> tweets) {
-            this.conf = conf;
+        public TweetsCreationTask(Tweeter tweeter, String searchText, BlockingQueue<Tweet> tweets) {
```
Comment (svenreimers) [inline 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java]: Evaluate: Should this be better as part of the new Tweet API abstraction layer (reduces duplication)?

File: 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java [hunk 4]
```diff
@@ -249,26 +214,21 @@ private TweetInfo checkNewTweetHasTags(TweetInfo info) {
 
     private class TweetsUpdateTask extends Task<Void> {
```
Comment (svenreimers) [inline 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java]: Evaluate: Should this be better as part of the new Tweet API abstraction layer (reduces duplication)?

File: 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java (hunk 3 above)
Comment (mklaehn) [inline 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java]: I was able to extract the TweetsCreationTask into the TweetSetData storage object keeping track of the underlying data being displayed in whatever view is desired

File: build.gradle (hunk 2 above)
Comment (mklaehn) [inline build.gradle]: Done

File: 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java (hunk 1 above)
Comment (mklaehn) [inline 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java]: Exception handling enhanced. Logging/NIL-behavior and with the ability to observe the latest exception

File: 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java (hunk 4 above)
Comment (mklaehn) [inline 2d/src/main/java/org/tweetwallfx/twod/TagTweets.java]: unable to abstract this as it is intertwined in with the actual view

**Pull request merged**